### PR TITLE
[examples] Bump @playwright/test to 0.1101.0

### PR DESCRIPTION
**Pull request by saltyshiomix**
Hi there, I love to use `@playwright/test`!

I think we need to update whole examples so I wrote this PR.

Thanks in advance.

Best,

### CODE REVIEW

File: examples/basic-js/package.json [hunk 1]
```diff
@@ -5,10 +5,10 @@
     "test": "tests"
   },
   "dependencies": {
-    "@playwright/test": "0.162.0"
+    "@playwright/test": "0.1101.0"
   },
   "scripts": {
-    "test": "folio"
+    "test": "folio --config=config.js"
```
Comment (mxschmitt) [inline examples/basic-js/package.json:11]: I would call all the configs `folio.config.ts` then you don't have to specify it manually with the --config parameter

File: examples/login-once-per-worker/config.ts [hunk 2]
```diff
@@ -14,14 +14,26 @@
...
+  testDir: __dirname,
+  timeout: 30000,
+});
+
+// New test type that provides a number in addition to standard page, context, etc.
+export const test = newTestType<{ loginOnce: (context: BrowserContext) => Promise<BrowserContext> }>();
 export { expect } from '@playwright/test';
 
-const builder = baseFolio.extend<{}, { loggedInState: any }>();
+let loggedInState: any;
```
Comment (mxschmitt) [inline examples/login-once-per-worker/config.ts:30]: instead of having the state here in a global variable, I would move it in an instance variable of LoggedInEnv. I would move the functions also directly into this as a private methods.

File: examples/screenshot-on-failure/tests/test.spec.ts [hunk 3]
```diff
@@ -14,14 +14,10 @@
...
 
-import { it, expect } from '@playwright/test';
+import { test, expect } from '@playwright/test';
 
-// Run this test with the '--param screenshotOnFailure' command line parameter
-// or 'npm run test'.
-
-it('is a basic test with the page', async ({ page, browserName, testInfo }) => {
+test('is a basic test with the page', async ({ page, browserName }) => {
   await page.setContent(`<div style="height: 500px; background-color: red">
```
Comment (mxschmitt) [inline examples/screenshot-on-failure/tests/test.spec.ts:20]:
```suggestion
test('is a basic test with the page', async ({ page, browserName }, testInfo) => {
  console.log(testInfo.title)
```
why did you remove this? You can get it like in the suggestion.

**Review by mxschmitt — COMMENTED** [review]
Looks good so far!

File: examples/login-once-per-worker/config.ts (hunk 2 above)
Comment (saltyshiomix) [reply examples/login-once-per-worker/config.ts:30]: Yes I tried to place the instance into `LoggedInEnv`, but the error `this.loggedInState is not defined` occurred.

Maybe `merge` function extracts only the specific property like this:

```ts
const { beforeEach } = LoggedInEnv;
```

File: examples/screenshot-on-failure/tests/test.spec.ts (hunk 3 above)
Comment (saltyshiomix) [reply examples/screenshot-on-failure/tests/test.spec.ts:20]: Because I can't find the `testInfo` in the second argument.

File: examples/basic-js/package.json (hunk 1 above)
Comment (saltyshiomix) [reply examples/basic-js/package.json:11]: Ok, I'll fix the examples :)

File: examples/screenshot-on-failure/tests/test.spec.ts (hunk 3 above)
Comment (saltyshiomix) [reply examples/screenshot-on-failure/tests/test.spec.ts:20]: Sorry, I misunderstood. I'll fix it soon.

File: examples/login-once-per-worker/config.ts (hunk 2 above)
Comment (mxschmitt) [reply examples/login-once-per-worker/config.ts]: I played a bit around, seems like that would work: https://gist.github.com/mxschmitt/85c18efe336d9c7307f10046459a4023

Comment (saltyshiomix) [reply examples/login-once-per-worker/config.ts]: Yes my expectation is more or less correct:

https://github.com/microsoft/folio/blob/2202f43ba956bed88c1b78464351ef40ed81b1e0/src/spec.ts#L47-L98

Comment (mxschmitt) [reply examples/login-once-per-worker/config.ts]: yeah agree, that's super hacky. Will try to follow up on folio or you if you want to fix it, feel free to open a PR there.

Comment (saltyshiomix) [reply examples/login-once-per-worker/config.ts]: Your solution works perfectly!

I fixed this in https://github.com/microsoft/playwright-test/pull/246/commits/bde971daeea61a880a5472c110efb70f726424e6

**Review by mxschmitt — APPROVED** [review]

**Comment by mxschmitt** [comment]
We decided to move this project to the main repository to benefit from the monorepo advantages which contains the examples now in the docs. But thanks for your contribution!
See here: https://playwright.dev/docs/test-intro
